core/src/Kokkos_Serial.hpp

```diff
@@ -120,9 +120,24 @@ class Serial {
   /// return asynchronously, before the functor completes.  This
   /// method does not return until all dispatched functors on this
   /// device have completed.
-  static void impl_static_fence() {}
+  static void impl_static_fence() {
+    impl_static_fence(
+        "Kokkos::Serial::impl_static_fence: Unnamed Static Fence");
+  }
+  static void impl_static_fence(const std::string& name) {
+    Kokkos::Tools::Experimental::Impl::profile_fence_event<Kokkos::Serial>(
+        name,
+        Kokkos::Tools::Experimental::SpecialSynchronizationCases::
+            GlobalDeviceSynchronization,
+        []() {});  // TODO: correct device ID
```

Comment:
What does "device ID" mean for `Serial`?

Reply:
Here I should just delete the comment, it's out-of-date. For Serial, you'll only see two device ID's, The first is the one you'll hit here. This is hit when somebody calls Kokkos::fence, and it's a fence of "the Serial Device." It's the semantic equivalent of a cudaDeviceSynchronize. The ID we provide there will be [the Serial DeviceType ID] << 24 (and as Serial = 0, this will be 0). The other will be for an instance fence. Now, in the Serial case, there's only one distinguishable instance, there's no "create a Serial instance from a Serial stream." So the TL;DR is that there will be only two instances: 0 and 1, for Serial.

Reply:
That might actually change. If you have external threads, for thread safety we may need to have multiple instances for example to have individual scratch pads.